### PR TITLE
activity-module

### PR DESCRIPTION
**Pull request by brandonschabell**
Implemented route tracking, pacing, distance tracking, and calorie counting.
Also contains multiDex support in the build.gradle file

### CODE REVIEW

File: Health/src/main/java/com/sciencesquad/health/activity/ActivityModule.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.sciencesquad.health.activity;
+
+public class ActivityModule {
```
Comment (avaidyam) [inline Health/src/main/java/com/sciencesquad/health/activity/ActivityModule.java:3]: lol well ok, fuck me then

File: Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        mMap.moveCamera(CameraUpdateFactory.newLatLng(latLng));
+    }
+
+    // DEFAULT WEIGHT (To be changed later)
+    double weightKG = 70;
+
+    public double calorieBurn(double speed, double timeDiff, double weightKG) {
+        // METS List (Metric for the exertion for each exercise)
+        double METS = 0;
+        double speedMPH = speed*2.23694; //converting speed from m/s to MPH
```
Comment (avaidyam) [inline Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:216]: There _must_ be a better way to do this than a bunch of if statements...

File: Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        double calories = METS * weightKG * timeDiff/3600;
+        return calories;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        switch (requestCode) {
+            case REQUEST_LOCATION_PERMISSION: {
+                if ((grantResults.length > 0) && (grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
+                    // Do what you wanted to do with the permissions
```
Comment (avaidyam) [inline Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:248]: So.... why is this here?

File: Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+            default:
+                return;
+        }
+    }
+
+    @Override
+    public void onConnected(Bundle bundle) {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION}, REQUEST_LOCATION_PERMISSION);
+            // here to request the missing permissions, and then overriding
```
Comment (avaidyam) [inline Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:262]: Why not do what the comment says?

Comment (brandonschabell) [inline Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:262]: that's what the thing above is doing

Comment (avaidyam) [inline Health/src/main/java/com/sciencesquad/health/activity/MapsActivity.java:262]: so very lost, but we'll go over it next time, no worries. :crown:

**Comment by colinrford** [comment]
Dear anyone looking, 
@brandonschabell said that there's a local file that doesn't sync with git. Due to this, Travis CI will fail. However, this branch works just fine locally for us (so it seems?????????)

**Comment by Brahmancer** [comment]
It looks like his branch is a few commits behind master. Maybe that's the reason?

**Comment by colinrford** [comment]
@Brahmancer we just merged master and resolved conflicts, I'm 99.9999% sure it's not behind master

**Comment by Brahmancer** [comment]
screen shot 2016-03-04 at 12 13 13 am

That's what I'm seeing.

**Comment by colinrford** [comment]
Github is pissing me off because we literally resolved all issues

**Comment by colinrford** [comment]
@Brahmancer you were right, for some reason it didn't show up for me? I assume I was doing something wrong then. This branch should be caught up now commit-wise.

**Comment by colinrford** [comment]
@avaidyam I'm going to leave this branch up to you to merge unless there's something I can do. Brandon said it wasn't going to pass Travis because of the Health365.jks file

**Comment by Brahmancer** [comment]
@colinrford were you able to build it on your machine?

**Comment by colinrford** [comment]
@Brahmancer I was able to build this on my computer, however, I did not test it on my phone. Brandon had it working, but perhaps we can double check with him before we proceed with merging

**Comment by Brahmancer** [comment]
I tried running this branch on my end and it gives me the thing below. As unfortunate/insecure as it is, we need that keystore in our repo. 
Error:Execution failed for task ':Health:validateConfigSigning'.

> Keystore file /Users/brandonschabell/keystores/Health365.jks not found for signing config 'config'.

**Comment by colinrford** [comment]
@brandonschabell suggested we each keep it locally and adjust the path accordingly

File: Health/build.gradle [hunk 5]
```diff
@@ -21,9 +21,16 @@ repositories {
...
 
 android {
+    signingConfigs {
+        config {
+            keyAlias 'Health365'
+            keyPassword 'd3012356101530'
+            storeFile file('/Users/brandonschabell/keystores/Health365.jks')
+            storePassword 'd3012356101530'
+        }
+    }
```
Comment (Brahmancer) [inline Health/build.gradle]: Trying removing this Part 1

**Comment by Brahmancer** [comment]
What that is implying is that each of us set up a gradle.properties in the ~/.gradle directory and create environment variables. Here's a link on how to set that up. 
http://stackoverflow.com/questions/18328730/how-to-create-a-release-signed-apk-file-using-gradle/21020469#21020469

However, if we do it that way, we need to set up a gradle.properties for Travis which I'm not sure how to do.

**Comment by colinrford** [comment]
That's something we should probably do, although I don't know if it should happen today. That's a good find though!

**Comment by Brahmancer** [comment]
For real though, with time constraints, not sure how to approach this situation without being insecure.

**Comment by colinrford** [comment]
We could leave this PR for when we're all around, or for Aditya to override. Then whoever demos the app or this aspect of the app will make sure they have the .jpk file n all that

**Comment by Brahmancer** [comment]
We'll talk about it more when we get together.

**Comment by avaidyam** [comment]
Override it is.
